tools/tcpaccept.py

```diff
@@ -161,29 +161,26 @@
     if (args->protocol != IPPROTO_TCP)
         return 0;
     u32 pid = bpf_get_current_pid_tgid();
-    struct sock *newsk = (struct sock *)args->skaddr;
-
     // pull in details
     u16 family = 0, lport = 0;
-    family = newsk->__sk_common.skc_family;
-    lport = newsk->__sk_common.skc_num;
+    family = args->family;
+    lport = args->sport;
 
+    // pull in details
     if (family == AF_INET) {
         struct ipv4_data_t data4 = {.pid = pid, .ip = 4};
         data4.ts_us = bpf_ktime_get_ns() / 1000;
-        data4.saddr = newsk->__sk_common.skc_rcv_saddr;
-        data4.daddr = newsk->__sk_common.skc_daddr;
+        bpf_probe_read(&data4.saddr, sizeof(u32), args->saddr);
+        bpf_probe_read(&data4.daddr, sizeof(u32), args->daddr);
```

Comment:
You don't need to use `bpf_probe_read` to read ctx members.

Reply:
how about "bpf_probe_read(&data6.saddr, sizeof(data6.saddr), args->saddr_v6)" ,
can direct use : data6.saddr = args->saddr_v6

Reply:
My bad. You'll actually need to use `bpf_probe_read` for both IPv4 and IPv6 addresses here since they're declared as arrays of `__u8`.

Reply:
Although it is an array, it is still directly accessible from ctx pointer so bpf_probe_read is not needed, you can do a __builtin_memcpy like below,
```
__builtin_memcpy(&data6.saddr, args->saddr_v6, sizeof(data6.saddr));
```
The clang compiler will turn it into two 8-byte load.